File: Back/Apis/Api_Js/src/models/tablas_sql/tabla_archivos_firmados.js
//Librerias 
import { DataTypes } from "sequelize";

//Configuraciones Creadas
import { conexion_base_datos } from "../../settings/consultas_sql/conexion_sql.js";

export const tabla_archivos_firmados = conexion_base_datos.define(
    "archivos_firmados",
    {
        archivo_id:{
            type: DataTypes.INTEGER,
            primaryKey: true,
            allowNull: false,
            autoIncrement: true,
        },
        archivo_direccion:{
            type: DataTypes.STRING,
            allowNull: false,
        },
        archivo_fecha:{
            type: DataTypes.DATE,
            allowNull: false,
        },
        usuario_id_fk:{
            type: DataTypes.INTEGER,
            allowNull: false,
        },
    },
    {
        freezeTableName: true,
        timestamps: false,
    }
)

File: Back/Apis/Api_Py/src/models/tablas_sql/tabla_archivos_firmados.py
#Librerias Instaladas
from sqlalchemy import ForeignKey, String, CheckConstraint;
from sqlalchemy.orm import Mapped, mapped_column, relationship;

#Tablas Creadas
from src.models.tablas_sql.tabla_base import Tabla_Base;

#Modulos Nativos
from typing import List, Optional;
from datetime import datetime;

class Tabla_Archivos_Firmados(Tabla_Base):
    __tablename__ = 'archivos_firmados';

    archivo_id: Mapped[int] = mapped_column(nullable=False, primary_key=True, autoincrement=True);
    archivo_direccion: Mapped[str] = mapped_column(String(1000),nullable=False);
    archivo_fecha: Mapped[datetime] = mapped_column(nullable=False);
    usuario_id_fk: Mapped[int] = mapped_column(ForeignKey('usuarios.usuario_id'), nullable=False);

    restriccion_usuarios_en_archivos: Mapped['Tabla_Usuarios'] = relationship(back_populates='restriccion_usuarios_para_archivos');

    def __repr__(self) -> str:
        return f"Tabla_Archivos_Firmados(archivo_id={self.archivo_id!r}, archivo_direccion={self.archivo_direccion!r}, archivo_fecha={self.archivo_fecha!r}, usuario_id_fk={self.usuario_id_fk!r})"

